refactor: Use `useSettings` hook in DefaultRedirectionSnackbar

Ldoppea opened the pull request:
In cozy/cozy-client#1460 we implemented a new `useSettings` hook

This hook can be used in `DefaultRedirectionSnackbar` in order to simplify code and homogenize the way we access cozy-apps settings

```
### ✨ Features

*

### 🐛 Bug Fixes

*

### 🔧 Tech

* Refactor `DefaultRedirectionSnackbar` to use new `useSettings` hook
```

zatteo commented on src/components/DefaultRedirectionSnackbar/helpers.js:
```diff
@@ -1,102 +1,43 @@
-    ...homeSettings,
-    default_redirection_view_count: default_redirection_view_count
-      ? default_redirection_view_count + 1
-      : 1
```
I do not see anymore this logic of setting 1 if there was no value before. Will it work ?

Ldoppea replied:
Fixed [here](https://github.com/cozy/cozy-home/compare/f17fe1ae5ddae709aec71980566f2dc9658dae08..7070fa5d79c09e42c6a980af74c0322d19930999) and in [cozy-client](https://github.com/cozy/cozy-client/pull/1460/commits/98cfaddd2f1e10ceb0d4259962aa4f473de95154)

Ldoppea replied:
Finally we did not include the default parameter option in the useSettings API, so I added the necessary code in this cozy-home PR

Ldoppea committed:
refactor: Use `useSettings` hook in DefaultRedirectionSnackbar

In cozy/cozy-client#1460 we implemented a new `useSettings` hook

This hook can be used in `DefaultRedirectionSnackbar` in order to
simplify code and homogenize the way we access cozy-apps settings

Ldoppea committed:
refactor: Rename `helper` to `useShouldShowDefaultRedirectionSnackbar`

Ldoppea committed:
feat: Upgrade cozy-client to 46.9.0

`cozy-client` has been upgraded to `46.9.0` to retrieve the new
useSettings hook

Related PR: cozy/cozy-client#1460
